src/search.rs

```diff
@@ -0,0 +1,147 @@
+use crate::document::advanced::DocumentEncryptUnmanagedResult;
+use crate::document::advanced::*;
+use crate::document::DocumentEncryptOpts;
+use crate::internal::{take_lock, IronOxideErr};
+use crate::{GroupId, IronOxide};
+use async_trait::async_trait;
+use rand::{self, RngCore};
+use std::convert::{TryFrom, TryInto};
+use std::ops::DerefMut;
+
+use ironcore_search_helpers::generate_hashes_for_string;
+
+type Result<A> = std::result::Result<A, IronOxideErr>;
+///The required length of the salt.
+const REQUIRED_LEN: usize = 32;
+
+///The result of creating a new index as well as initializing an IronSimpleSdk.
+///If you only want to create the index, see create_index.
+pub struct CreatedIndexResult {
+    pub encrypted_salt: DocumentEncryptUnmanagedResult,
+    pub sdk: IronSimpleSearch,
+}
+
+///Trait which gives the ability to create an index.
+#[async_trait]
+pub trait SimpleSeachInitialize {
+    ///Given the encrypted salt and the edeks, decrypt them and give back the IronSimpleSearch object.
+    async fn initialize_search_sdk(
+        &self,
+        encrypted_data: &[u8],
```

Comment:
Should we just call this the `encrypted_salt`?

Reply:
I changed it to encrypted_salt_bytes to hopefully avoid confusion between the entire "EncryptedSalt" which is the deks + bytes and just the Aes bytes.